exonum/src/blockchain/service.rs

```diff
@@ -32,13 +32,22 @@ use blockchain::{Blockchain, ConsensusConfig, Schema, StoredConfiguration, Valid
 use helpers::{Height, Milliseconds, ValidatorId};
 use super::transaction::Transaction;
 
-/// A trait that describes business logic of a concrete service.
+/// A trait that describes the business logic of a certain service.
+///
+/// Services are the main extension point for the Exonum framework. Initially,
+/// Exonum does not provide specific transaction processing rules or business
+/// logic, they are implemented with the help of services.
+///
+/// The code above indicates the other traits on which the `Service` trait is
+/// dependent and the functions required for this trait. The first four functions
```

Comment:
I don't think that such a strict description of functions order is a good idea. Those lines may change easily and it's not obvious for the developer that some comment below may rely on it. In result we can get erroneous documentation.

And is it really "code above"? Kind of confusing.